uaclient/entitlements/base.py

```diff
@@ -128,10 +145,53 @@ def valid_service(self):
 
         return self._valid_service
 
-    @abc.abstractmethod
+    def check_for_reboot_msg(self, operation: str) -> None:
+        """Check if user should be alerted that a reboot must be performed.
+
+        @param operation: The operation being executed.
+        """
+        if util.should_reboot():
+            print(
+                status.MESSAGE_ENABLE_REBOOT_REQUIRED_TMPL.format(
+                    operation=operation
+                )
+            )
+
     def enable(self, *, silent_if_inapplicable: bool = False) -> bool:
         """Enable specific entitlement.
 
+        :param silent_if_inapplicable:
+            Don't emit any messages until after it has been determined that
+            this entitlement is applicable to the current machine.
+
+        @return: True on success, False otherwise.
+        """
+        msg_ops = self.messaging.get("pre_enable", [])
+        if not handle_message_operations(msg_ops):
+            return False
+        can_enable, _ = self.can_enable(silent=silent_if_inapplicable)
+        if not can_enable:
+            return False
+
+        ret = self._perform_enable(
+            silent_if_inapplicable=silent_if_inapplicable
+        )
+        if not ret:
+            return False
+
+        msg_ops = self.messaging.get("post_enable", [])
+        if not handle_message_operations(msg_ops):
+            return False
+
```

Comment:
I left that in the `RepoEntitlement` because it didn't seem general enough to be called for every entitlement.  (Also it was causing some bizarre behavior in tests when I tried putting it here, like `os.path.exists` returning true for a file that doesn't exist...)

Reply:
Sorry, I think the method is now on `base.py`. If we bring it back to `RepoEntitlement`, no problem at all

Reply:
Oof! Good catch, yep I played around with moving it which is how that happened. I'll move it back